src/SFA.DAS.Commitments.Infrastructure/Data/JobProgressRepository.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Data;
+using System.Threading.Tasks;
+using Dapper;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Interfaces;
+using SFA.DAS.Sql.Client;
+
+namespace SFA.DAS.Commitments.Infrastructure.Data
+{
+    public class JobProgressRepository : BaseRepository, IJobProgressRepository
+    {
+        private readonly ICommitmentsLogger _logger;
+
+        public JobProgressRepository(string connectionString, ICommitmentsLogger logger)
+            : base(connectionString, logger.BaseLogger)
+        {
+            _logger = logger;
+        }
+
+        // column per progress tag, or row per tag?
+        // column +ve : tags can be different data type
+        // row +ve : unlimited tags, no schema change for extra tags unless require different data type
+        //     -ve : tags all same data type, or muliple columns
+        // https://stackoverflow.com/questions/3967372/sql-server-how-to-constrain-a-table-to-contain-a-single-row
+
+        public async Task<long?> Get_AddEpaToApprenticeships_LastSubmissionEventId()
+        {
+            _logger.Debug("Getting last (processed by AddEpaToApprenticeships) SubmissionEvents Id");
+
+            return await WithConnection(async connection => await connection.ExecuteScalarAsync<long?>(
+                "SELECT [AddEpa_LastSubmissionEventId] FROM [dbo].[JobProgress]",
+                commandType: CommandType.Text));
+        }
+
+        public async Task Set_AddEpaToApprenticeships_LastSubmissionEventId(long lastSubmissionEventId)
+        {
+            _logger.Debug($"Setting last (processed by AddEpaToApprenticeships) SubmissionEvent Id to {lastSubmissionEventId}");
+
+            await WithConnection(async connection =>
+            {
+                var parameters = new DynamicParameters();
+                parameters.Add("@lastSubmissionEventId", lastSubmissionEventId, DbType.Int64);
+
+                return await connection.ExecuteAsync(
```

Comment:
Total new SQL to me! I feel total out of date

Reply:
Use with caution :-)
https://www.mssqltips.com/sqlservertip/3074/use-caution-with-sql-servers-merge-statement/